src/TeamSyncer.ts

```diff
@@ -273,15 +280,24 @@ export class TeamSyncer {
         }
 
         // Create the room first.
+        let roomId: string;
         try {
-            const roomId = await this.createRoomForChannel(teamId, channelItem.creator, channelItem);
+            roomId = await this.createRoomForChannel(teamId, channelItem.creator, channelItem);
             await this.main.actionLink({
                 matrix_room_id: roomId,
                 slack_channel_id: channelItem.id,
                 team_id: teamId,
             });
         } catch (ex) {
             log.error("Failed to provision new room dynamically:", ex);
+            return;
+        }
+
+        try {
+            await this.syncMembershipForRoom(roomId, channelItem.id, teamId, client);
```

Comment:
In general it doesn't seem correct to have have two calls to `syncMembershipForRoom`.

the logic is:

```
if (! existingroom) {
  // creation steps
}
syncMembershipForRoom()
```

But it seems like `creation steps` is so large it's better the way you have it. maybe 

```
if (! existingroom) {
  creationSteps()
}
syncMembershipForRoom()

function creationSteps() {
  // creation steps
}
```

could work, but not worth refactoring rn.